avocado/core/runner.py

```diff
@@ -466,9 +466,9 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
             TEST_LOG.error('Job interrupted by ctrl+c.')
             summary.add('INTERRUPTED')
 
-        self.result.end_tests()
-        self.job.funcatexit.run()
         if self.job.sysinfo is not None:
             self.job.sysinfo.end_job_hook()
+        self.result.end_tests()
+        self.job.funcatexit.run()
```

Comment:
So, I was thinking about this... It's a tough call.  It really depends on how he define and document the `funcatexit` feature.  Right now, this is what we say about them:

"It’s possible to register a callback function that will be called when all the tests have finished running"

So it's accurate in that sense.  All tests have finished running... and IMHO, having the results generated are part of that in a greater sense.

So, right now, I'd leave it as is.